systemtest/src/test/java/io/strimzi/systemtest/bridge/HttpBridgeKafkaExternalListenersST.java

```diff
@@ -213,6 +216,9 @@ private void testWeirdUsername(ExtensionContext extensionContext, String weirdUs
             .withTopicName(topicName)
             .withMessageCount(MESSAGE_COUNT)
             .withUserName(weirdUserName)
+            // we disable ssl.endpoint.identification.algorithm for external listener (i.e., Nodeport),
+            // because TLS hostname verification is not supported on such listener type.
+            .withAdditionalConfig("ssl.endpoint.identification.algorithm=\n")
```

Comment:
Same here ... Why does this use Node port listener here?

Reply:
The producer fails on this error:
```java
eadyOps=0] SSL Handshake failed
javax.net.ssl.SSLHandshakeException: No subject alternative names matching IP address 192.168.0.226 found
	at sun.security.ssl.Alert.createSSLException(Alert.java:131) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:353) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:296) ~[?:?]
	at sun.security.ssl.TransportContext.fatal(TransportContext.java:291) ~[?:?]
	at sun.security.ssl.CertificateMessage$T13CertificateConsumer.checkServerCerts(CertificateMessage.java:1357) ~[?:?]
	at sun.security.ssl.CertificateMessage$T13CertificateConsumer.onConsumeCertificate(CertificateMessage.java:1232) ~[?:?]
	at sun.security.ssl.CertificateMessage$T13CertificateConsumer.consume(CertificateMessage.java:1175) ~[?:?]
	at sun.security.ssl.SSLHandshake.consume(SSLHandshake.java:392) ~[?:?]
	at sun.security.ssl.HandshakeContext.dispatch(HandshakeContext.java:443) ~[?:?]
	at sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1074) ~[?:?]
	at sun.security.ssl.SSLEngineImpl$DelegatedTask$DelegatedAction.run(SSLEngineImpl.java:1061) ~[?:?]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at sun.security.ssl.SSLEngineImpl$DelegatedTask.run(SSLEngineImpl.java:1008) ~[?:?]
```

Reply:
I'm not wondering about the disabling of it. I'm wondering why does this use node-ports in the first place as it seems unnecessary.

Reply:
I am not sure...you mean we should rather produce messages to the `internal listener` instead of producing them via external NodePort? 

Because these test cases currently do the following:
1. Create a bridge consumer to receive messages from the external listener (NodePort) 
2. Create a producer to produce messages via the same listener (NodePort)

So should we modify the second point to use an internal listener instead of an external one?

Reply:
Yeah, I think we should think about what the purpose of the test is. i mean I can approve this and we can merge it. But we should look into that in some separate PR - what exactly does this bring us.

Reply:
IIRC reason why we are using external producer etc. was because at the time I was writing those "weird username tests", there were no possibilities to use internal clients - as they didn't allow any username to be longer than some number of characters and it shouldn't contain any special characters.

External clients are allowing it. 

TBH I don't remember the exact reason why we went this way, but maybe we can rewrite those test to use our test-clients.

But the rewriting itself should be for a different PR.

Reply:
> Yeah, I think we should think about what the purpose of the test is.

I agree, we definitely could do an investigation into such test cases and find out if they are necessary. :) I can create an issue for that.